apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/Operations.java

```diff
@@ -240,51 +235,10 @@ public ExpireSnapshots.Result expireSnapshots(Table table, long expireBeforeTime
    */
   public void runRetention(
       String fqtn, String columnName, String columnPattern, String granularity, int count) {
-    checkRecords(fqtn, columnName, columnPattern);
     final String statement =
         SparkJobUtil.createDeleteStatement(fqtn, columnName, columnPattern, granularity, count);
-    Dataset<Row> dsWithExpiredRows =
-        spark.sql(
-            SparkJobUtil.createSelectLimitStatement(
-                fqtn, columnName, columnPattern, granularity, count, 1));
-    if (!dsWithExpiredRows.isEmpty()) {
-      spark.sql(statement).show();
-    }
-  }
-
-  /**
-   * CheckRecords verifies a sample from fqtn against @columnPattern to validate parsing with
-   * columnPattern. The metrics emitted helps to verify if records in table are not adhering to
-   * DateTimeFormat standards
-   *
-   * @param fqtn - fully-qualified table name
-   * @param columnName - retention column name
-   * @param columnPattern - retention column pattern
-   */
-  public void checkRecords(String fqtn, String columnName, String columnPattern) {
-    String quotedFqtn = SparkJobUtil.getQuotedFqtn(fqtn);
-    if (!StringUtils.isBlank(columnPattern)) {
-      String query =
-          String.format(
-              "SELECT COALESCE(to_timestamp(%s, '%s'), 'Invalid') AS parsed_timestamp FROM %s LIMIT 10",
-              columnName, columnPattern, quotedFqtn);
-      log.info("Pattern verification query {}", query);
-      List<Row> resultDF = spark.sql(query).collectAsList();
-      if (resultDF.stream()
-          .map(r -> r.getString(r.fieldIndex("parsed_timestamp")))
-          .collect(Collectors.toList())
-          .contains("Invalid")) {
-        meter
-            .counterBuilder(AppConstants.INCOMPATIBLE_DATE_COLUMN)
-            .build()
-            .add(1, Attributes.of(AttributeKey.stringKey(AppConstants.TABLE_NAME), fqtn));
-        log.warn(
-            "Failed to parse column {} with provided retention column pattern {} for table {}",
-            columnName,
-            columnPattern,
-            fqtn);
-      }
-    }
+    Dataset<Row> dsWithExpiredRows = spark.sql(statement);
+    log.info("Retention Job deleted records: {} from table: {}", dsWithExpiredRows.count(), fqtn);
```

Comment:
this is going to be expensive itself ?

Reply:
Collecting count of deleted records should be fairly quick and is not part of Query itself.
Holding the list of records consumes some memory but should be in MBs for some 10 * (X) * 1000 records.

Reply:
Would this be metadata-only operation or will have to scan removed files?

Reply:
It's not clear if collectAsList is only metadata only ops. I don't have sufficient proof for it.
Removing it as there is cost associated with adding logs. Having deleted record count helps while debugging to check an deletion is happening with each retention job. 
I guess this gap can be filled by performing some analysis from table stats published to kafka.

Thanks for double clicking on it.